### PR TITLE
FC Networking: Added NetworkingLinkLoginWarmup pane

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR adds the initial code for NetworkingLinkLoginWarmup. It also adds support for adding an e-mail in the Example App Playground.

## Testing

Simulator Screen Shot - iPhone 14 Pro - 2023-02-07 at 10 58 18

https://user-images.githubusercontent.com/105514761/217297159-ddfde132-e356-44be-b58e-ff71646d182d.mov

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupBodyView.swift [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+final class NetworkingLinkLoginWarmupBodyView: UIView {
+
+    private let didSelectContinue: () -> Void
+
+    init(
+        email: String,
+        didSelectContinue: @escaping (() -> Void),
+        didSelectSkip: @escaping (() -> Void)
+    ) {
+        self.didSelectContinue = didSelectContinue
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupBodyView.swift:23]: so didSelectSkip will be retained by the label's action?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupDataSource.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ) {
+        self.manifest = manifest
+        self.apiClient = apiClient
+        self.clientSecret = clientSecret
+        self.analyticsClient = analyticsClient
+    }
+
+    func disableNetworking() -> Future<Void> {
+        return apiClient.disableNetworking(
+            disabledReason: nil,
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupDataSource.swift:39]: would we pass a different reason in some cases?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupViewController.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            ),
+            subtitle: STPLocalizedString(
+                "It looks like you have a Link account. Signing in will let you quickly access your saved bank accounts.",
+                "The subtitle/description of a screen where users are informed that they can sign-in-to Link."
+            ),
+            contentView: NetworkingLinkLoginWarmupBodyView(
+                // `accountholderCustomerEmailAddress` should always be non-null, and
+                // since the email is only used as a visual, it's not worth to throw an error
+                // if it is null
+                email: dataSource.manifest.accountholderCustomerEmailAddress ?? "you",
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupViewController.swift:55]: good one

**Review by vardges-stripe — APPROVED** [review]
nice stuff. some questions

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupDataSource.swift (hunk 2 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupDataSource.swift:39]: @vardges-stripe yes there is [one case](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/10f7ffc3/src/linkedAccounts/inner/components/panes/LinkLogin/useSaveToNetworkAndLink.ts#L124) in Stripe.js, but this particular case does not have a disabled reason 

(note that the strict following of Stripe.js [and the avoidance of extra in-depth research] is intentional for shipping speed)

**Pull request merged**